lib/iamspec/action/generic_action.rb

```diff
@@ -1,18 +1,33 @@
 module Iamspec::Action
   def perform_action(action_name, resource_arns = [])
-    GenericAction.new([action_name], resource_arns)
+    GenericAction.new([action_name], nil, resource_arns)
   end
+
+  def perform_action_with_caller_arn(action_name, caller_arn, resource_arns = [])
+    GenericAction.new([action_name], caller_arn, resource_arns)
+  end
+
   def perform_actions(action_names, resource_arns = [])
-    GenericAction.new(action_names, resource_arns)
+    GenericAction.new(action_names, nil, resource_arns)
   end
 
   class GenericAction
     attr_reader :action_names
+    attr_reader :caller_arn
     attr_reader :resource_arns
+    attr_reader :creds
+    attr_reader :policy_string
+    attr_reader :policy
+    attr_reader :userid
+    attr_reader :sourcevpce
+    attr_reader :sourceip
+    attr_reader :context_entries
 
-    def initialize(action_names, resource_arns)
+    def initialize(action_names, caller_arn, resource_arns)
```

Comment:
new parameter introduced here but not added in the `AssumeRole` Action that calls the `super` constructor: 
```
 1) AWS IAM Integration Using syntactic sugar for users IAM User "some_user_with_admin_permissions"
     Failure/Error:
       def initialize(action_names, caller_arn, resource_arns)
         @caller_arn = caller_arn
         @action_names = action_names
         @resource_arns = resource_arns
         @context_entries = {}
       end

     ArgumentError:
       wrong number of arguments (2 for 3)
     # ./lib/iamspec/action/generic_action.rb:26:in `initialize'
     # ./lib/iamspec/action/assume_role.rb:11:in `initialize'
     # ./lib/iamspec/action/assume_role.rb:5:in `new'
     # ./lib/iamspec/action/assume_role.rb:5:in `assume_role'
     # ./spec/integration_spec.rb:23:in `block (5 levels) in <top (required)>'

  2) AWS IAM Integration Using syntactic sugar for users IAM User "some_user_without_admin_permissions"
     Failure/Error:
       def initialize(action_names, caller_arn, resource_arns)
         @caller_arn = caller_arn
         @action_names = action_names
         @resource_arns = resource_arns
         @context_entries = {}
       end

     ArgumentError:
       wrong number of arguments (2 for 3)
     # ./lib/iamspec/action/generic_action.rb:26:in `initialize'
     # ./lib/iamspec/action/assume_role.rb:11:in `initialize'
     # ./lib/iamspec/action/assume_role.rb:5:in `new'
     # ./lib/iamspec/action/assume_role.rb:5:in `assume_role'
     # ./spec/integration_spec.rb:27:in `block (5 levels) in <top (required)>'
```

Reply:
Fixed